In [2]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import plotly.express as px

In [3]:
df = pd.read_csv("./forbes_billionaires.csv")

df.head()

rank                      name        forbes_id  net_worth   age  \
0     1  Bernard Arnault & family  bernard-arnault      211.0  74.0   
1     2                 Elon Musk        elon-musk      180.0  51.0   
2     3                Jeff Bezos       jeff-bezos      114.0  59.0   
3     4             Larry Ellison    larry-ellison      107.0  78.0   
4     5            Warren Buffett   warren-buffett      106.0  92.0   

   age_range        country              source               industry   Age  \
0          8         France                LVMH       Fashion & Retail  74.0   
1          6  United States       Tesla, SpaceX             Automotive  52.0   
2          6  United States              Amazon             Technology  59.0   
3          8  United States              Oracle             Technology  79.0   
4         10  United States  Berkshire Hathaway  Finance & Investments  92.0   

   ...           Residence    Citizenship      Marital Status Children  \
0  ...       Paris, France         France             Married      5.0   
1  ...       Austin, Texas  United States              Single     10.0   
2  ...  Medina, Washington  United States             Engaged      4.0   
3  ...       Lanai, Hawaii  United States     In Relationship      4.0   
4  ...     Omaha, Nebraska  United States  Widowed, Remarried      3.0   

                                           Education Bachelor  Master  \
0  Bachelor of Arts/Science, Ecole Polytechnique ...      1.0     0.0   
1  Bachelor of Arts/Science, University of Pennsy...      1.0     0.0   
2     Bachelor of Arts/Science, Princeton University      1.0     0.0   
3  Drop Out, University of Chicago; Drop Out, Uni...      0.0     0.0   
4  Master of Science, Columbia University; Bachel...      1.0     1.0   

  Doctorate  Drop Out  Self Made  
0       0.0       0.0          0  
1       0.0       0.0          1  
2       0.0       0.0          1  
3       0.0       1.0          1  
4       0.0       0.0          1  

[5 rows x 23 columns]

In [ ]:
df.isnull().sum()

rank                     0
name                     0
forbes_id                0
net_worth                0
age                     65
age_range                0
country                  0
source                   0
industry                 0
Age                     85
Source of Wealth         1
Self-Made Score       2082
Philanthropy Score    2229
Residence               59
Citizenship             11
Marital Status         551
Children              1042
Education             1225
Bachelor              1224
Master                1224
Doctorate             1224
Drop Out              1224
Self Made                0
dtype: int64

In [ ]:
df.shape

(2640, 23)

In [ ]:
df.industry.unique()

array(['Fashion & Retail', 'Automotive', 'Technology',
       'Finance & Investments', 'Media & Entertainment', 'Telecom',
       'Diversified', 'Food & Beverage', 'Logistics',
       'Gambling & Casinos', 'Manufacturing', 'Real Estate',
       'Metals & Mining', 'Energy', 'Healthcare', 'Service',
       'Construction & Engineering', 'Sports'], dtype=object)

In [ ]:
df.country.unique()

array(['France', 'United States', 'Mexico', 'India', 'Spain', 'China',
       'Canada', 'Germany', 'Italy', 'Hong Kong', 'Austria', 'Japan',
       'Switzerland', 'Australia', 'Indonesia', 'Russia', 'Chile',
       'United Kingdom', 'Israel', 'Brazil', 'Czech Republic',
       'Singapore', 'Sweden', 'Thailand', 'Netherlands', 'Nigeria',
       'Cyprus', 'Malaysia', 'United Arab Emirates', 'South Africa',
       'New Zealand', 'Philippines', 'Monaco', 'Belgium', 'South Korea',
       'Taiwan', 'Norway', 'Egypt', 'Denmark', 'Ireland',
       'Eswatini (Swaziland)', 'Colombia', 'Poland', 'Ukraine', 'Greece',
       'Turkey', 'Argentina', 'Georgia', 'Portugal', 'Kazakhstan',
       'Algeria', 'Venezuela', 'Vietnam', 'Finland', 'Belize', 'Lebanon',
       'Oman', 'Iceland', 'Guernsey', 'Liechtenstein', 'Bulgaria',
       'Romania', 'Zimbabwe', 'Qatar', 'Nepal', 'Slovakia', 'Morocco',
       'Hungary', 'Tanzania', 'Peru', 'Barbados', 'Macau', 'Estonia',
       'St. Kitts and Nevis', 'Armenia

## Wealth Distribution by Country:

In [ ]:
df.country.value_counts()

United States           735
China                   494
India                   169
Germany                 126
Russia                  104
                       ... 
Belize                    1
Eswatini (Swaziland)      1
Venezuela                 1
Algeria                   1
Panama                    1
Name: country, Length: 77, dtype: int64

In [ ]:
fig = px.pie(df, values=df.country.value_counts()[:10], names=df.country.value_counts()[:10].index[:10], title='Percentage of billionaire in countries', )
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()

Here from the data we can observe that united states has highest number of billionaires with count of **735** and it accounts for total **38.2%** of global billionaire count. Then china ranks second with **494** billionaires and third is india with **169** billionaires.


---





## Top 10 billionaires by net worth

In [ ]:
top_ten_df = df.nlargest(10, ['net_worth']).sort_values(by='net_worth', ascending=True)

top_ten_df

rank                       name          forbes_id  net_worth   age  \
9    10              Steve Ballmer      steve-ballmer       80.7  67.0   
8     9              Mukesh Ambani      mukesh-ambani       83.4  65.0   
7     8  Carlos Slim Helu & family   carlos-slim-helu       93.0  83.0   
6     7          Michael Bloomberg  michael-bloomberg       94.5  81.0   
5     6                 Bill Gates         bill-gates      104.0  67.0   
4     5             Warren Buffett     warren-buffett      106.0  92.0   
3     4              Larry Ellison      larry-ellison      107.0  78.0   
2     3                 Jeff Bezos         jeff-bezos      114.0  59.0   
1     2                  Elon Musk          elon-musk      180.0  51.0   
0     1   Bernard Arnault & family    bernard-arnault      211.0  74.0   

   age_range        country              source               industry   Age  \
9          7  United States           Microsoft             Technology  67.0   
8          7          India         Diversified            Diversified  66.0   
7          9         Mexico             Telecom                Telecom  83.0   
6          9  United States        Bloomberg LP  Media & Entertainment  81.0   
5          7  United States           Microsoft             Technology  67.0   
4         10  United States  Berkshire Hathaway  Finance & Investments  92.0   
3          8  United States              Oracle             Technology  79.0   
2          6  United States              Amazon             Technology  59.0   
1          6  United States       Tesla, SpaceX             Automotive  52.0   
0          8         France                LVMH       Fashion & Retail  74.0   

   ...                Residence    Citizenship      Marital Status Children  \
9  ...  Hunts Point, Washington  United States             Married      3.0   
8  ...            Mumbai, India          India             Married      3.0   
7  ...      Mexico City, Mexico         Mexico             Widowed      6.0   
6  ...       New York, New York  United States     In Relationship      2.0   
5  ...       Medina, Washington  United States            Divorced      3.0   
4  ...          Omaha, Nebraska  United States  Widowed, Remarried      3.0   
3  ...            Lanai, Hawaii  United States     In Relationship      4.0   
2  ...       Medina, Washington  United States             Engaged      4.0   
1  ...            Austin, Texas  United States              Single     10.0   
0  ...            Paris, France         France             Married      5.0   

                                           Education Bachelor  Master  \
9  Bachelor of Arts/Science, Harvard University; ...      1.0     0.0   
8  Drop Out, Stanford University; Bachelor of Sci...      1.0     0.0   
7  Bachelor of Arts/Science, Universidad Nacional...      1.0     0.0   
6  Master of Business Administration, Harvard Bus...      1.0     1.0   
5                       Drop Out, Harvard University      0.0     0.0   
4  Master of Science, Columbia University; Bachel...      1.0     1.0   
3  Drop Out, University of Chicago; Drop Out, Uni...      0.0     0.0   
2     Bachelor of Arts/Science, Princeton University      1.0     0.0   
1  Bachelor of Arts/Science, University of Pennsy...      1.0     0.0   
0  Bachelor of Arts/Science, Ecole Polytechnique ...      1.0     0.0   

  Doctorate  Drop Out  Self Made  
9       0.0       1.0          1  
8       0.0       1.0          0  
7       0.0       0.0          1  
6       0.0       0.0          1  
5       0.0       1.0          1  
4       0.0       0.0          1  
3       0.0       1.0          1  
2       0.0       0.0          1  
1       0.0       0.0          1  
0       0.0       0.0          0  

[10 rows x 23 columns]

In [ ]:
fig = px.bar(top_ten_df, y='name', x='net_worth', color='net_worth', title='Top 10 Billionaires by Net Worth',color_continuous_scale='darkmint' ,text_auto=True)
fig.update_layout(xaxis_title='Net Worth (in billions)',
                  yaxis_title='Billionaire Name')

fig.show()

Here is list of top 10 billionaires by net worth.

From the graph we can observe that **Bernard Arnault & family** tops with \$211 billion. Second is Elon Musk with \$180 billion net worth.



---





## Number of Self-Made and Non-Self-Made Billionaires

In [ ]:
df['Self Made'].value_counts()

1    1812
0     828
Name: Self Made, dtype: int64

In [ ]:
self_made_counts = df['Self Made'].value_counts()

# Create a DataFrame for the counts
counts_df = pd.DataFrame({'Self Made': self_made_counts.index, 'Count': self_made_counts.values})

df["Self Made"] = df["Self Made"].astype(str)

# Create the bar plot using Plotly Express
fig = px.bar(counts_df, x='Self Made', y='Count', color='Self Made',
             title='Number of Self-Made and Non-Self-Made Billionaires', text_auto=True)

# Customize the layout
fig.update_layout(xaxis_title='Billionaire Type',
                  yaxis_title='Count',
                  showlegend=False)  # Hide the legend since color represents the categories
# customise text
fig.update_traces(textfont_size=12, textangle=0, textposition="outside")

# Show the plot
fig.show()

From the data we can see that there are 1812 self made billionaires and 828 non self made billionaires.

---



## Number of billionaires in each industry

In [ ]:
df.industry.value_counts()

Finance & Investments         372
Manufacturing                 324
Technology                    314
Fashion & Retail              266
Food & Beverage               212
Healthcare                    201
Real Estate                   193
Diversified                   187
Energy                        100
Media & Entertainment          91
Metals & Mining                74
Automotive                     73
Service                        53
Construction & Engineering     45
Logistics                      40
Sports                         39
Telecom                        31
Gambling & Casinos             25
Name: industry, dtype: int64

In [ ]:
industry_billionaire = df.industry.value_counts();

# make data frame

industry_df = pd.DataFrame({'Industry' : industry_billionaire.index, 'Number of billionaire': industry_billionaire.values}).sort_values(by='Number of billionaire', ascending=False)

# bar plot for number of billionaire in each industry

fig = px.bar(industry_df, y = 'Industry', x = 'Number of billionaire', title="Number of Billionaire in each industry", text_auto=True, color='Industry', color_discrete_sequence = px.colors.qualitative.Set3)
fig.update_layout(showlegend=False, height=800)

# show graph

fig.show()

From the graph we can see that there are highest number of billionaires in Finance and Investment industry with total 372 billionaires. Second highest industry is Manufacturing with total 324 billionaires. Third industry is Technology with 314 billionaires.

---


In [ ]:
industry_df

Industry  count
0        Finance & Investments    372
1                Manufacturing    324
2                   Technology    314
3             Fashion & Retail    266
4              Food & Beverage    212
5                   Healthcare    201
6                  Real Estate    193
7                  Diversified    187
8                       Energy    100
9        Media & Entertainment     91
10             Metals & Mining     74
11                  Automotive     73
12                     Service     53
13  Construction & Engineering     45
14                   Logistics     40
15                      Sports     39
16                     Telecom     31
17          Gambling & Casinos     25

## Number of billionaires by age

In [ ]:
df.age.value_counts()

60.0     87
58.0     83
59.0     82
57.0     75
66.0     74
         ..
20.0      1
19.0      1
18.0      1
101.0     1
28.0      1
Name: age, Length: 79, dtype: int64

In [11]:
df['age'].sort_values().head()

823     18.0
1755    19.0
1334    20.0
825     21.0
1754    21.0
Name: age, dtype: float64

In [9]:
df['age'].sort_values(ascending=False).head()

2180    101.0
1398     99.0
1342     99.0
2327     99.0
145      99.0
Name: age, dtype: float64

In [ ]:
fig = px.histogram(df, x="age", text_auto=True)
fig.update_traces(textfont_size=12, textangle=0, textposition="outside")
fig.show()

From the above graph we can see that there are most number of billionaires of age 58-59 with count of 165 billionaires. Youngest billionaire is of age 18. Oldest billionaire is of age 101.

---


## Top 20 cities by number of billionaires.

In [ ]:
city_series = df['Residence'].value_counts();

# creating city dataframe

city_df = pd.DataFrame({'city': city_series.index, 'count': city_series.values})

city_df


city  count
0         New York, New York     98
1             Beijing, China     68
2       Hong Kong, Hong Kong     67
3            Shanghai, China     63
4     London, United Kingdom     60
..                       ...    ...
767      Montpellier, France      1
768  Santa Clara, California      1
769       Stuttgart, Germany      1
770    Indianapolis, Indiana      1
771      Makati, Philippines      1

[772 rows x 2 columns]

In [ ]:
city_df_top_20 = city_df.nlargest(20, ['count']).sort_values('count', ascending=True)

fig = px.bar(city_df_top_20, y='city', x='count',  text_auto = True)

fig.show();

City with the highest number of billionaires is New York. There are total 98 billionaires in New York. Second highest is Beijing, China with total 68 billionaires. Third is Hong Kong with total 67 billionaires

## Counts of Self-Made Billionaires by Country

In [ ]:
selfmade = df[df['Self Made'] == 1]

# counts of self made billionaires in each country

selfmade_counts = selfmade['Citizenship'].value_counts()

# Create a DataFrame for the counts
selfmade_counts_df = pd.DataFrame({'Country': selfmade_counts.index, 'Count': selfmade_counts.values})

selfmade_counts_df = selfmade_counts_df.nlargest(25, 'Count')

selfmade_counts_df

Country  Count
0    United States    517
1            China    480
2           Russia    103
3            India     73
4           Canada     48
5   United Kingdom     47
6        Hong Kong     42
7          Germany     35
8           Taiwan     34
9        Australia     32
10           Japan     31
11           Italy     22
12          Brazil     22
13          Israel     21
14     Switzerland     21
15       Singapore     20
16          France     18
17          Sweden     18
18       Indonesia     17
19        Thailand     15
20          Turkey     15
21     South Korea     13
22           Spain     10
23         Austria      9
24  Czech Republic      9

In [ ]:
fig = px.bar(selfmade_counts_df, x='Country', y='Count',
             title='Counts of Self-Made Billionaires by Country',
             labels={'Country': 'Country', 'Count': 'Count of Billionaires'},
             text_auto=True)

fig.update_traces(textposition= 'outside')
fig.show()

This graph represents countries with highest the number of self made billionaires. United States is the at rank 1. There are total 517 self made billionaires. Second is China with 480 self made billionaires. Third country is Russia with total 103 billionaires. India is at rank 4 with 73 billionaires

---

In [ ]:
selfmade

rank                    name       forbes_id  net_worth   age  \
1        2               Elon Musk       elon-musk      180.0  51.0   
2        3              Jeff Bezos      jeff-bezos      114.0  59.0   
3        4           Larry Ellison   larry-ellison      107.0  78.0   
4        5          Warren Buffett  warren-buffett      106.0  92.0   
5        6              Bill Gates      bill-gates      104.0  67.0   
...    ...                     ...             ...        ...   ...   
2633  2540               Yan Junxu       yan-junxu        1.0  53.0   
2634  2540   Yi Xianzhong & family    yi-xianzhong        1.0  63.0   
2635  2540                 Yu Rong         yu-rong        1.0  51.0   
2637  2540           Zhang Gongyun   zhang-gongyun        1.0  60.0   
2638  2540  Zhang Guiping & family   zhang-guiping        1.0  71.0   

      age_range        country                        source  \
1             6  United States                 Tesla, SpaceX   
2             6  United States                        Amazon   
3             8  United States                        Oracle   
4            10  United States            Berkshire Hathaway   
5             7  United States                     Microsoft   
...         ...            ...                           ...   
2633          6          China                 Manufacturing   
2634          7          China               Pharmaceuticals   
2635          6          China                Health clinics   
2637          6          China  Tyre manufacturing machinery   
2638          8          China                   Real estate   

                   industry   Age  ...           Residence    Citizenship  \
1                Automotive  52.0  ...       Austin, Texas  United States   
2                Technology  59.0  ...  Medina, Washington  United States   
3                Technology  79.0  ...       Lanai, Hawaii  United States   
4     Finance & Investments  92.0  ...     Omaha, Nebraska  United States   
5                Technology  67.0  ...  Medina, Washington  United States   
...                     ...   ...  ...                 ...            ...   
2633          Manufacturing  54.0  ...      Taicang, China          China   
2634             Healthcare  64.0  ...    Guangzhou, China          China   
2635             Healthcare  51.0  ...     Shanghai, China          China   
2637          Manufacturing  60.0  ...        Gaomi, China          China   
2638            Real Estate  72.0  ...      Nanjing, China          China   

          Marital Status Children  \
1                 Single     10.0   
2                Engaged      4.0   
3        In Relationship      4.0   
4     Widowed, Remarried      3.0   
5               Divorced      3.0   
...                  ...      ...   
2633             Married      NaN   
2634                 NaN      NaN   
2635                 NaN      NaN   
2637                 NaN      NaN   
2638             Married      1.0   

                                              Education Bachelor  Master  \
1     Bachelor of Arts/Science, University of Pennsy...      1.0     0.0   
2        Bachelor of Arts/Science, Princeton University      1.0     0.0   
3     Drop Out, University of Chicago; Drop Out, Uni...      0.0     0.0   
4     Master of Science, Columbia University; Bachel...      1.0     1.0   
5                          Drop Out, Harvard University      0.0     0.0   
...                                                 ...      ...     ...   
2633                                                NaN      NaN     NaN   
2634                                                NaN      NaN     NaN   
2635                                                NaN      NaN     NaN   
2637                                                NaN      NaN     NaN   
2638    Bachelor of Arts/Science, South East University      1.0     0.0   

     Doctorate  Drop Out  Self Made  
1          0.0       0.0          1  
2          0.0       0.0 

## Self Made vs Non Self Made billionaire in India

In [ ]:
india_data = df[df['country'] == 'India']

india_data

rank                       name                  forbes_id  net_worth  \
8        9              Mukesh Ambani              mukesh-ambani       83.4   
23      24               Gautam Adani             gautam-adani-1       47.2   
54      55                 Shiv Nadar                 shiv-nadar       25.6   
67      68           Cyrus Poonawalla           cyrus-poonawalla       22.6   
92      93             Lakshmi Mittal             lakshmi-mittal       17.7   
...    ...                        ...                        ...        ...   
2498  2405              Rajesh Sharma              rajesh-sharma        1.1   
2539  2540        Manohar Lal Agarwal        manohar-lal-agarwal        1.0   
2557  2540  Kochouseph Chittilappilly  kochouseph-chittilappilly        1.0   
2580  2540          Inder Jaisinghani        inder-jaisinghani-1        1.0   
2584  2540              Ashok Kajaria              ashok-kajaria        1.0   

       age  age_range country                       source  \
8     65.0          7   India                  Diversified   
23    60.0          6   India  Infrastructure, commodities   
54    77.0          8   India            software services   
67    81.0          9   India                     Vaccines   
92    72.0          8   India                        Steel   
...    ...        ...     ...                          ...   
2498  53.0          6   India                      Finance   
2539  68.0          7   India                       Snacks   
2557  72.0          8   India        Electrical appliances   
2580  70.0          7   India               Cables & wires   
2584  75.0          8   India                        Tiles   

                   industry   Age  ...               Residence  Citizenship  \
8               Diversified  66.0  ...           Mumbai, India        India   
23              Diversified  61.0  ...        Ahmedabad, India        India   
54               Technology  78.0  ...            Delhi, India        India   
67               Healthcare  82.0  ...             Pune, India        India   
92          Metals & Mining  73.0  ...  London, United Kingdom        India   
...                     ...   ...  ...                     ...          ...   
2498  Finance & Investments  53.0  ...           Mumbai, India        India   
2539        Food & Beverage  68.0  ...            Delhi, India        India   
2557          Manufacturing  72.0  ...            Kochi, India        India   
2580          Manufacturing  70.0  ...           Mumbai, India        India   
2584          Manufacturing  76.0  ...            Delhi, India        India   

      Marital Status Children  \
8            Married      3.0   
23           Married      2.0   
54           Married      1.0   
67           Widowed      1.0   
92           Married      2.0   
...              ...      ...   
2498         Married      3.0   
2539         Married      NaN   
2557         Married      2.0   
2580         Married      2.0   
2584         Married      2.0   

                                              Education Bachelor  Master  \
8     Drop Out, Stanford University; Bachelor of Sci...      1.0     0.0   
23                                                  NaN      NaN     NaN   
54    Bachelor of Arts/Science, PSG College of Techn...      1.0     0.0   
67    Bachelor of Arts/Science, Pune University; Doc...      1.0     0.0   
92    Bachelor of Arts/Science, St Xavier's College ...      1.0     0.0   
...                                                 ...      ...     ...   
2498  Chartered Accountant, Institute of Chartered A...      0.0     0.0   
2539                                                NaN      NaN     NaN   
2557                                                NaN      NaN     NaN   
2580                                                NaN      NaN     NaN   
2584           Bachelor of Science in Engineering, UCLA      1.0     0.0   

     Doctorate  Drop Out  Self Made  
8          0.0 

In [ ]:
indian_selfmade = pd.DataFrame({
    'Self made': ['Non Self Made', 'Self Made'],
    'Count': india_data['Self Made'].value_counts().values
})

indian_selfmade

Self made  Count
0  Non Self Made     96
1      Self Made     73

In [ ]:
fig = px.pie(indian_selfmade, values=indian_selfmade['Count'], names=indian_selfmade['Self made'], title='Self Made vs Non Self Made billionaire in India')
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()

This graph represents self made vs non self made billionaires in india.

There are more non self made billionaires in india compared to self made billionaires. Total count of non self made billionaires is 96 and count of self made billionaires is 73.

---

## Top 20 billionaires in India

In [ ]:
top_indians = pd.DataFrame({'Name': india_data['name'], 'Net Worth': india_data['net_worth'], 'Self Made': india_data['Self Made']}).nlargest(20, 'Net Worth').sort_values(by = 'Net Worth', ascending=True)

top_indians

Name  Net Worth  Self Made
457       Vikram Lal & family        5.6          0
429         Gopikishan Damani        6.0          1
422              Sunil Mittal        6.1          1
338       Rohiqa Cyrus Mistry        7.0          0
335      Ashwin Dani & family        7.0          0
311         Benu Gopal Bangur        7.3          0
290  Hasmukh Chudgar & family        7.6          1
252          Kushal Pal Singh        8.2          0
233             Ravi Jaipuria        8.6          0
211               Azim Premji        9.2          0
137                Uday Kotak       12.9          1
123               Kumar Birla       14.2          0
113        Radhakishan Damani       15.3          1
111            Dilip Shanghvi       15.6          1
95    Savitri Jindal & family       17.5          0
92             Lakshmi Mittal       17.7          0
67           Cyrus Poonawalla       22.6          0
54                 Shiv Nadar       25.6          1
23               Gautam Adani       47.2          1
8               Mukesh Ambani       83.4          0

In [ ]:
fig = px.bar(top_indians,
             y='Name',
             x='Net Worth',
             color='Self Made',
             color_continuous_scale='temps',
             text_auto=True,
             title='Top 20 billionaires in India')
fig.update_layout(height=800)
fig.update_traces(textposition = 'outside')
fig.show()

This graph represents the top 20 billionaires of India by net worth and also indicating that are they self made or non self made billionaire.

Mukesh Ambani is the richest person in india. His net worth is \$83.4 billion and he is non self made billionaire.

Gautam Adani is second richest person in India with net worth of \$47.2 billion. He is self made billionaire. Third richest person is Shiv Nadar with net worth of \$25.6 billion and he is self made billionaire.

---

## Number of billionaires in each industry in india

In [ ]:
# Count the number of billionaires in each industry
industry_counts = india_data['industry'].value_counts()

# Create a DataFrame for the counts
counts_df = pd.DataFrame({'industry': industry_counts.index, 'Count': industry_counts.values})

# Create the bar plot using Plotly Express
fig = px.bar(counts_df, x='industry', y='Count',
             title='Number of Billionaires in Each Industry in India',
             labels={'industry': 'Industry', 'Count': 'Count of Billionaires'},
             color='industry')

# Customize the layout
fig.update_layout(xaxis_title='Industry', yaxis_title='Count')

# Rotate x-axis labels for better readability
fig.update_xaxes(tickangle=45)

# Show the plot
fig.show()

## Self made billionaire in each industry

In [ ]:
self_made_billionaires = df[['industry', 'Self Made']]

# self_made_billionaires = self_made_billionaires['Self Made']

self_made_billionaires

industry Self Made
0          Fashion & Retail         0
1                Automotive         1
2                Technology         1
3                Technology         1
4     Finance & Investments         1
...                     ...       ...
2635             Healthcare         1
2636        Food & Beverage         0
2637          Manufacturing         1
2638            Real Estate         1
2639            Diversified         0

[2640 rows x 2 columns]

In [ ]:
self_made_billionaires = df[df['Self Made'] == '1']

# Count the number of self-made billionaires in each industry
self_made_industry_counts = self_made_billionaires['industry'].value_counts()

# Create a DataFrame for the counts
counts_df = pd.DataFrame({'Industry': self_made_industry_counts.index, 'Count': self_made_industry_counts.values})

# Create the bar plot using Plotly Express
fig = px.bar(counts_df, x='Industry', y='Count',
             title='Number of Self-Made Billionaires in Each Industry',
             labels={'Industry': 'Industry', 'Count': 'Count of Self-Made Billionaires'},
             color = 'Industry')

# Customize the layout
fig.update_layout(xaxis_title='Industry', yaxis_title='Count')

# Rotate x-axis labels for better readability
fig.update_xaxes(tickangle=45)

# Show the plot
fig.show()